packages/model/DimensionMetadata.ts

```diff
@@ -4,14 +4,23 @@ import { schema } from '@tpluscode/rdf-ns-builders'
 import { cc } from '@cube-creator/core/namespace'
 import { initializer } from './lib/initializer'
 
-interface DimensionMetadata {
+export interface DimensionMetadata extends RdfResource {
   about: NamedNode
 }
 
 export interface DimensionMetadataCollection extends RdfResource {
   hasPart: Array<DimensionMetadata>
 }
 
+export function DimensionMetadataMixin<Base extends Constructor>(Resource: Base) {
+  class Impl extends Resource implements Partial<DimensionMetadata> {
+    @property.resource({ path: schema.about })
+    about!: NamedNode
+  }
+
+  return Impl
+}
+
```

Comment:
re. the CI error, look you will need to annotate the intended target for this mixin. This one does not have a class associated nor any other way to use it implicitly. Maybe try this

```suggestion
export function DimensionMetadataMixin<Base extends Constructor>(Resource: Base) {
  class Impl extends Resource implements Partial<DimensionMetadata> {
    @property.resource({ path: schema.about })
    about!: NamedNode
  }

  return Impl
}

DimensionMetadataMixin.shouldApply = false
```